helper/exptime/backoff.go

```diff
@@ -0,0 +1,142 @@
+// Package exptime provides a generalized exponential backoff retry implementation.
```

Comment:
We implement "exponential back-off" in a few places, but none of them generic and feature-full. It'd be nice to converge on a single implementation.